src/celeritas/optical/PhotonPrimary.hh

```diff
@@ -0,0 +1,30 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/PhotonPrimary.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Types.hh"
+#include "orange/Types.hh"
+#include "celeritas/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Optical photon data used to initialize a photon track state.
+ */
+struct PhotonPrimary
```

Comment:
Maybe we want to call it `OpticalPhotonPrimary`?

Reply:
Either `OpticalPhotonPrimary` or just `OpticalPhoton` should be good.  OpticalPhotonPrimary may be better as `OpticalPhoton` may stand for the particle definition for optical photons.

Reply:
I was thinking something similar after our discussions but worry that `OpticalPhotonPrimary` is too unwieldy. Perhaps we can abbreviate `OptPhoPrimary` or is that ugly? Maybe `OpticalPrimary`? We're going to have a lot of stuff in `celeritas/optical` that otherwise will get prefixes.

Reply:
I like `OpticalPrimary`.